keps/sig-auth/5284-limit-impersonate-permission/README.md

```diff
@@ -0,0 +1,985 @@
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+<!--
+**Note:** This checklist is iterative and should be reviewed and updated every time this enhancement is being considered for a milestone.
+-->
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+This is to add additional access control over the existing impersonation action. An impersonator
+who impersonates another user is required to have the additional permissions to impersonate on
+certain group resources and verbs.
+
+## Motivation
+Today an impersonator can impersonate another user if the impersonator has the permission of
+```yaml
+- verbs: ["impersonate"]
+  resources: ["users"]
+  resourceNames: ["someUser"]
+```
+However, current impersonation mechanism is the "unrestricted impersonation", the impersonator can
+get more power through impersonation. And because of this, it is strongly suggested not
+to allow the controller to impersonate. More detailed discussion is in this
+[issue](https://github.com/kubernetes/kubernetes/issues/27152)
+
+There are use cases that needs a controller to be able to impersonate:
+1. A controller impersonates a node the controller is running on. This happens when 
+per-node agents (CNI plugins for instance), want to read pods on a given node instead
+of unrestricted pod access.
+2. A controller runs as a deputy, receives request from the user, and impersonates
+the user to start/stop virtual machine or access vm console managed by kubevirt.
+
+This proposal is to introduce additional permissions for impersonation, so that any 
+impersonator can impersonate in a more restricted way.
+
+### Goals
+* Define required permission rules for the impersonator to impersonate 
+  on certain resource/verb.
+* Provide clear example on what permission the impersonator and user should have for the
+  impersonator to impersonate user's certain verbs on certain resources
+
+### Non-Goals
+* Adding impersonator's info to user.Info, so authorization webhooks can know both info
+of impersonated user and the requester.
+
+## Proposal
+
+Introduce a set of verbs with prefix of `impersonate-on:`, e.g. `impersonate-on:create` and
+`impersonate-on:get`. The impersonator needs to have these verbs with certain resources to
+impersonate. 
+
+Introduce verbs `impersonate:user` and `impersonate:scheduled-node`:
+- `impersonate:user` limits the impersonator to impersonate users with 
+certain names/groups/userextras. The resources must be `users`/`groups`/`userextras`.
+The resource names must be user names, group names or values in the user extras accoringly.
+- `impersonate:scheduled-node` that limits the impersonator to impersonate the node the
```

Comment:
Include the exact set of impersonation headers the workload must set for the node case.